lib/cylc/task_job_mgr.py

```diff
@@ -760,6 +759,8 @@ def _prep_submit_task_job(self, suite, itask, dry_run, check_syntax=True):
             itask.summary['job_hosts'][itask.submit_num] = ''
             # Retry delays, needed for the try_num
             self._set_retry_timers(itask, rtconfig)
+            # Create job dir now because we're about to return
+            self._create_job_log_path(suite, itask)
```

Comment:
Took me a while to figure out this had to be before the next line otherwise the job-activity.log wouldn't get created, even though the dir it goes in would get created. Still don't know why